crates/revm/src/builder.rs

```diff
@@ -195,6 +195,36 @@ impl<'a, EXT, DB: Database> EvmBuilder<'a, HandlerStage, EXT, DB> {
         }
     }
 
+    /// This modifies the [EvmBuilder] to make it easy to construct an [`Evm`] with a _specific_
+    /// handler.
+    ///
+    /// # Example
+    /// ```rust
+    /// use revm::{EvmBuilder, Handler, primitives::{SpecId, HandlerCfg}};
+    /// use revm_interpreter::primitives::CancunSpec;
+    /// let builder = EvmBuilder::default();
+    /// // set up empty handler cfg
+    /// let handler = HandlerCfg::new(SpecId::LATEST);
+    /// let builder = builder.with_handler_cfg(handler);
+    ///
+    /// // get the desired handler
+    /// let mainnet = Handler::mainnet::<CancunSpec>();
+    /// let builder = builder.with_handler(mainnet);
+    ///
+    /// // build the EVM
+    /// let evm = builder.build();
+    /// ```
+    pub fn with_handler(
```

Comment:
This function is inside:
`impl<'a, EXT, DB: Database> EvmBuilder<'a, HandlerStage, EXT, DB> {`

But it should be in:
`impl<'a, BuilderStage, EXT, DB: Database> EvmBuilder<'a, BuilderStage, EXT, DB> {`

Reason for this is that `with_handler` can be called from both `HandlerStage` and `SetGenericStage `

Reply:
ah I see, yeah it makes sense to call it from either handler or setgeneric stages